core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1528,7 +1528,9 @@ private void handleOffsetFetch(@Nullable SqlNode offset, @Nullable SqlNode fetch
           ((SqlBasicCall) call).setOperator(overloads.get(0));
         }
       }
-      if (config.callRewrite()) {
+      if (config.callRewrite()
```

Comment:
This modification makes this logic less pure. I assume that when callRewrite is set to true, the original author intends to rewrite all expressions, not just under specific conditions. We should ensure that the SQL can still be executed after the rewrite, except when including nondeterministic functions, where there might be incorrect results.

Reply:
I have filed https://issues.apache.org/jira/browse/CALCITE-7012